AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -261,6 +260,42 @@ - (void)createWithPartition:(NSString *)partition
                                }];
 }
 
+- (void)deleteDocumentWithPartition:(NSString *)partition
+                         documentId:(NSString *)documentId
+                       writeOptions:(MSWriteOptions *)__unused writeOptions
+                  completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
+  // TODO consume writeOptions
+  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
+                   partitions:@[ partition ]
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+              // If error getting token.
+              if (error || !tokenResponses) {
+                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
+                completionHandler([[MSDataSourceError alloc] initWithError:error]);
+                return;
+              }
+
+              MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+              // Call CosmosDB
+              [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:cosmosDbIngestion
+                                                          tokenResult:tokenResponses.tokens[0]
```

Comment:
Question: does it make sense to check `tokenResponses.tokens.length > 0` in [line 272](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1399/files#diff-775692dad5e01b9791f9c25c9341fb31R272) and handles it there, or `performCosmosDbAsyncOperationWithHttpClient:tokenResult:documentId:httpMethod` takes care of nil tokens?

Reply:
I think it makes more sense to check in line 272. Check added